Architecture_Engine/Modify/TransformProfile.cs

```diff
@@ -0,0 +1,111 @@
+ */
+
+using BH.oM.Geometry;
+using BH.oM.Architecture.Theatron;
+using System;
+using BH.Engine.Base;
+using System.Linq;
+using BH.Engine.Geometry;
+using System.Collections.Generic;
+using System.ComponentModel;
+using BH.oM.Reflection.Attributes;
+
+namespace BH.Engine.Architecture.Theatron
+{
+    public static partial class Modify
+    {
+        [Description("Scale, rotate and translate a TierProfile")]
+        [Input("originalSection", "TierProfile to transform")]
+        [Input("scale", "Scaling amount")]
+        [Input("source", "Origin point for the transform")]
+        [Input("target", "Target point for the transform")]
+        [Input("angle", "Rotation angle")]
+        [PreviousVersion("4.2", "BH.Engine.Architecture.Theatron.Create.TransformProfile(BH.oM.Architecture.Theatron.TierProfile, BH.oM.Geometry.Vector, BH.oM.Geometry.Point, BH.oM.Geometry.Point, System.Double)")]
+        public static TierProfile TransformProfile(TierProfile originalSection, Vector scale, Point source, Point target, double angle)
+        {
+            if (originalSection == null)
+            {
+                BH.Engine.Reflection.Compute.RecordError("Cannot modify a null tier profile.");
+                return originalSection;
+            }
+
+            if(scale == null)
+            {
+                BH.Engine.Reflection.Compute.RecordError("Cannot transform a tier profile using a null scale vector.");
+                return originalSection;
+            }
+
+            if(source == null)
+            {
+                BH.Engine.Reflection.Compute.RecordError("Cannot transform a tier profile from a null source point.");
+                return originalSection;
+            }
+
+            if(target == null)
+            {
+                BH.Engine.Reflection.Compute.RecordError("Cannot transform a tier profile from a null target point.");
+                return originalSection;
+            }
+
+            TierProfile transformedTier = (TierProfile)originalSection.DeepClone();
+            var xScale = Geometry.Create.ScaleMatrix(source, scale);
+            var xRotate = Geometry.Create.RotationMatrix(source, Vector.ZAxis, angle);
+            var xTrans = Geometry.Create.TranslationMatrix(target - source);
+            TransformTier(ref transformedTier, xScale);
+            TransformTier(ref transformedTier, xRotate);
+            TransformTier(ref transformedTier, xTrans);
+            return transformedTier;
+        }
+
+        /***************************************************/
+
+        public static ProfileOrigin DefineTierOrigin(List<Point> flrPoints)
+        {
+            ProfileOrigin profOrigin = Create.ProfileOrigin(flrPoints[0], flrPoints[1] - flrPoints[0]);
+            return profOrigin;
+        }
+
+        /***************************************************/
+
+        public static void TransformTier(ref TierProfile profile, TransformMatrix xTrans)
```

Comment:
I would keep this method private to avoid mess with `ref` method parameters in the UI - assuming that neither of us has a solid grip over what is going on here @FraserGreenroyd 😄

Reply:
My main issue with this one is that these methods are used both in modify methods and create methods. I guess the easiest way is actually to avoid trying to fix the compliance of a create in modify and let @rolyhudson sort it out given it's his ~~mess~~ code 😝 he'll know best how to resolve it I imagine. @rolyhudson I'm happy to help with versioning, but following discussion with @pawelbaran I am going to revert the attempt to resolve that bit of compliance for now.